webapp/passport/js/protein.js

```diff
@@ -224,6 +218,176 @@ protein =
+                    plotData.push({
+                        timepoint: 'Before',
+                        intensity: this.peptides[i].beforeintensity,
+                        sequence: this.peptides[i].sequence,
+                        xLabel: abbreviatedSequence
+                    });
+                    plotData.push({
+                        timepoint: 'After',
+                        intensity: this.peptides[i].normalizedafterintensity ? this.peptides[i].normalizedafterintensity : 0.0,
+                        sequence: this.peptides[i].sequence,
+                        xLabel: abbreviatedSequence
+                    });
+                }
+
+                var groupedBoxData = LABKEY.vis.groupData(plotData, function (row) {
+                    return row.sequence
+                });
+
+                var medians = {};
+                var cvLineData = [];
+                var cvs = {};
+
+                for (var groupName in groupedBoxData) {
+                    var stats = LABKEY.vis.Stat.summary(groupedBoxData[groupName], function (row) {
+                        return row.intensity
+                    });
+                    var sd = LABKEY.vis.Stat.getStdDev(stats.sortedValues);
+                    medians[groupName] = stats.Q2;
+                    var cv = stats.Q2 ? (sd / stats.Q2) : null;
+                    cvLineData.push({x: groupName, cv: cv, color: 'median'});
+                    cvs[groupName] = cv;
+                }
+
+                var sortBy = this.settings.getSortBy();
+                if (sortBy === "Sequence Location") {
+                    plotData.sort(function (a, b) {
+                        return a["StartIndex"] - b["StartIndex"];
+                    });
+                }
+
+                if (sortBy === "Intensity") {
+                    plotData.sort(function (a, b) {
+                        return medians[b.sequence] - medians[a.sequence];
+                    });
+                }
+
+                if (sortBy === "Coefficient of Variation") {
+                    plotData.sort(function (a, b) {
+                        return cvs[b.sequence] - cvs[a.sequence];
+                    });
+                }
+
+                var cvBarLayer = new LABKEY.vis.Layer({
+                    geom: new LABKEY.vis.Geom.BarPlot({
+                        opacity: 0.2,
+                        hoverText: function (x, stats) {
+                            console.log('hover');
+                            return 'test';
+                        }
+                    }),
+                    aes: {
+                        x: 'x',
+                        yRight: 'cv',
+
+                    },
+                    data: cvLineData
+                });
+
+                var pointLayer = new LABKEY.vis.Layer({
+                    geom: new LABKEY.vis.Geom.Point({
+                        position: 'jitter',
+                        color: 'timepoint',
```

Comment:
is this color value?

Reply:
This tells the plot to use the value of the timepoint field in the data rows to assign colors to the plots.

![image](https://user-images.githubusercontent.com/5114642/109339872-22503a80-781d-11eb-98a9-1adbdd999323.png)

Reply:
got it.